### PR TITLE
Hotfixes

### PR DESCRIPTION
**Pull request by jupe**
This contains fixes from @kuggenhoffen -fork..

Related to: https://github.com/ARMmbed/socketIO-client/pull/1

### CODE REVIEW

**Comment by jupe** [comment]
@kuggenhoffen @VeliMattiLahtela @RomanSaveljev please review

File: websocket/_abnf.py [hunk 1]
```diff
@@ -354,6 +357,7 @@ def recv_frame(self):
         frame = ABNF(fin, rsv1, rsv2, rsv3, opcode, has_mask, payload)
         frame.validate(self.skip_utf8_validation)
 
+        self.lock.release()
```
Comment (RomanSaveljev) [inline websocket/_abnf.py:360]: release lock in finally block?

Comment (RomanSaveljev) [reply websocket/_abnf.py:360]: This could actually help to the situation we are having

**Comment by VeliMattiLahtela** [comment]
flashing does not work correctly when this is installed, one execution runs fine, but i.e. --repeat 5 will fail at flashing. With the version from pip it works fine. Well, flashing it self works fine, but the message that nothing was done is lost.

**Comment by kuggenhoffen** [comment]
This might not work with socketio-client since this is newer version than what it is expecting so need to rebase to 0.35 to test this fix

**Comment by jupe** [comment]
CC: @juhhov

File: websocket/_abnf.py [hunk 2]
```diff
@@ -307,6 +309,7 @@ def recv_mask(self):
         self.mask = self.recv_strict(4) if self.has_mask() else ""
 
     def recv_frame(self):
+        self.lock.acquire()
```
Comment (jupe) [inline websocket/_abnf.py:333]: @juhhov could you try this fix also?

**Comment by jupe** [comment]
Looks that similar fixes was done already to master..

**Pull request closed without merging**